projects/Benchmarks/Networking/Networking_BasicDeliver.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+
+namespace Benchmarks.Networking
+{
+    [MemoryDiagnoser]
+    public class Networking_BasicDeliver
+    {
+        private const int messageCount = 10000;
+
+        private IDisposable _container;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _container = RabbitMQBroker.Start(); 
+        }
+
+        [GlobalCleanup]
+        public void GlobalCleanup()
+        {
+            _container.Dispose();
+        }
+
+
+        [Benchmark(Baseline = true)]
+        public async Task Publish_Hello_World()
+        {
+            var cf = new ConnectionFactory { ConsumerDispatchConcurrency = 2 };
+            using (var connection = cf.CreateConnection())
+            {
+                await Publish_Hello_World(connection);
+            }
+        }
+
+        public static async Task Publish_Hello_World(IConnection connection)
+        {
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            using (var model = connection.CreateModel())
+            {
+                var queue = model.QueueDeclare();
+                var consumed = 0;
+                var consumer = new EventingBasicConsumer(model);
+                consumer.Received += (s, args) =>
+                {
+                    if (Interlocked.Increment(ref consumed) == messageCount)
+                    {
+                        tcs.SetResult(true);
+                    }
+                };
+                model.BasicConsume(queue.QueueName, true, consumer);
+
+                const string publish1 = "hello world";
+                byte[] body = Encoding.UTF8.GetBytes(publish1);
```

Comment:
Worth to move into `GlobalSetup` ?

Reply:
when we get to the unit of allocated bytes but as long we are at the range of MB it matters little where we allocate the 11 byte payload